exonum-java-binding-core/rust/src/utils/time_service.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use exonum::blockchain::Blockchain;
+
+use jni::objects::JClass;
+use jni::sys::jboolean;
+use jni::JNIEnv;
+
+use parking_lot::{Once, ONCE_INIT};
+
+const TIME_SERVICE_ID: u16 = 4;
+
+static INIT: Once = ONCE_INIT;
+static mut BLOCKCHAIN: Option<Blockchain> = None;
+
+#[no_mangle]
+pub extern "system" fn Java_com_exonum_binding_time_TimeSchemaImpl_isTimeServiceEnabled(
+    _: JNIEnv,
+    _: JClass,
+) -> jboolean {
+    is_time_service_enabled().into()
+}
+
+/// Initializes static context with reference to blockchain
+pub fn init_with_blockchain(blockchain: Blockchain) {
+    unsafe {
+        INIT.call_once(|| BLOCKCHAIN = Some(blockchain));
+    }
+}
+
+fn is_time_service_enabled() -> bool {
+    INIT.state().done() &&
+    unsafe { BLOCKCHAIN.clone() }.map_or(false, |blockchain|{
+        blockchain.service_map().get(&TIME_SERVICE_ID).is_some()
```

Comment:
Unfortunately this won't work properly in some scenarios... As service IDs are not fixed in Exonum, any user can create an own service with SERVICE_ID=4, hence this check will return true. 🤔

Reply:
Yes, this is rather an open topic on how to check it. Since the user is able to provide the same name as for time service, probably it worth checking the exact type (if possible) as well.

Reply:
But they are fixed in the services we include in the application, aren't they (configuration, btc-anchoring & time)? If so, it might be reasonable to forbid users to set such ids for their services.